libs/clients/data-protection-complaint/src/lib/data-protection-complaint-client.middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { Inject, Injectable } from '@nestjs/common'
+import {
+  FetchParams,
+  Middleware,
+  RequestContext,
+  SecurityApi,
+} from '../gen/fetch/dev'
+import {
+  CLIENT_CONFIG,
+  DataProtectionComplaintClientConfig,
+} from './data-protection-complaint-client.config'
+
+@Injectable()
+export class TokenMiddleware implements Middleware {
+  constructor(
+    @Inject(CLIENT_CONFIG)
+    private readonly config: DataProtectionComplaintClientConfig,
+    private readonly securityApi: SecurityApi,
+  ) {}
+
+  async pre(context: RequestContext): Promise<FetchParams | void> {
+    const responseToken = await this.securityApi.authenticate({
+      password: this.config.password,
+      username: this.config.username,
+    })
+
+    const token =
+      responseToken.startsWith('"') && responseToken.endsWith('"')
+        ? responseToken.substring(1, responseToken.length - 1)
+        : responseToken
```

Comment:
Can we ask someone at Personuvernd to change their API response to not include the double quotes?